psh/pshapp/pshapp.c

```diff
@@ -480,47 +483,103 @@ static int psh_cmpname(const void *n1, const void *n2)
 
 static int psh_keyCode(const char *buff, int *pEsc)
 {
-	int n, esc = *pEsc;
+	size_t n;
+	int esc = *pEsc;
+
+	/* ESC sequences ( ^[ ) */
+	static const struct {
+		char escChar;
+		int keyCode;
+	} escKeys[] = {
+		{ 'f', kAltRight },
+		{ 'b', kAltLeft },
+		{ 'd', kAltD },
+	};
+
+	/* CSI / SS3 sequences ( ^[[ / ^[O ) */
 	static const struct {
 		const char *escSeq;
 		int keyCode;
-	} keys[] = {
+	} csiKeys[] = {
 		{ "A", kUp },
 		{ "B", kDown },
 		{ "C", kRight },
 		{ "D", kLeft },
 		{ "F", kEnd },
 		{ "H", kHome },
+		{ "1;5C", kCtrlRight },
+		{ "1;5D", kCtrlLeft },
+		{ "1;5A", kCtrlUp },
+		{ "1;5B", kCtrlDown },
 		{ "1~", kHome },
 		{ "3~", kDelete },
 		{ "4~", kEnd },
 		{ "7~", kHome },
-		{ "8~", kEnd }
+		{ "8~", kEnd },
+		{ "1;3C", kAltRight },
+		{ "1;3D", kAltLeft },
+		{ "1;3A", kAltUp },
+		{ "1;3B", kAltDown },
 	};
 
-	if (esc < 4)
+	if (esc < 3) {
 		return -EAGAIN;
+	}
+
+	if ((buff[0] == '^') && (buff[1] == '[') && (buff[2] != '[') && (buff[2] != 'O')) {
+		for (n = 0U; n < (sizeof(escKeys) / sizeof(escKeys[0])); ++n) {
+			if (buff[2] == escKeys[n].escChar) {
+				return escKeys[n].keyCode;
+			}
+		}
+
+		return -EINVAL;
+	}
+
+	if (esc < 4) {
+		return -EAGAIN;
+	}
 
 	if (buff[0] == '^' && buff[1] == '[' && (buff[2] == '[' || buff[2] == 'O')) {
 		buff += 3;
 		esc -= 3;
 
-		for (n = 0; n < sizeof(keys) / sizeof(keys[0]); ++n) {
-			if (strncmp(buff, keys[n].escSeq, esc) != 0)
+		for (n = 0U; n < sizeof(csiKeys) / sizeof(csiKeys[0]); ++n) {
+			if (strncmp(buff, csiKeys[n].escSeq, esc) != 0)
```

Comment:
Missing brackets (now rest of the function code style has been updated)

Reply:
Done